filebeat/docs/how-does-filebeat-work.asciidoc

```diff
@@ -2,68 +2,68 @@
 
-=== What is a harvester?
+=== What is a reader?
 
-A harvester is responsible to harvester the content of a single file. That is done by reading each file line by line and send the information to the output. Each harvester is started by its prospector and retrieves all configuration options from the prospector. The harvester is responsible to open and close files. That means, as long as a harvester is running, the file descriptor stays open. That means, even if the file is removed or renamed, filebeat will keep reading the file. This has the side affect that the space on your disk will be reserved until the harvester is stopped.
+A reader is responsible to read the content of a single file. That is done by reading each file line by line and send the information to the output. Each reader is started by its input and retrieves all configuration options from the input. The reader is responsible to open and close files. That means, as long as a reader is running, the file descriptor stays open. That means, even if the file is removed or renamed, filebeat will keep reading the file. This has the side affect that the space on your disk will be reserved until the harvester is stopped.
 
-=== What is the Registry
+=== What is the state?
 
-Filebeat has a global registry to persist offset / state it was last reading from. This is to ensure all log lines are sent. In case Elasticsearch or Logstash are down, Filebeat will persist which were the last lines sent for each file and will continue reading the file, as soon elasticsearch becomes available again. As long as filebeat is running, all this information is also kept in memory but frequently written to disk. In case of a restart of filebeat, the data from the registry file is used to rebuild the state and filebeat continues at the last known position.
+Filebeat keeps the state of each file and frequently flushs it to disk. The state is used to remember the last offset a reader was reading from and to ensure all log lines are sent. In case Elasticsearch or Logstash are down, Filebeat will keep the track of the last lines sent for each file and will continue reading the file, as soon elasticsearch becomes available again. As long as filebeat is running, all this information is also kept in memory but frequently written to disk to the `registry_file`. In case of a restart of filebeat, the data from the registry file is used to rebuild the state and filebeat continues at the last known position.
 
-In the registry file, each havested file has an entry. As files can be renamed or moved, the filename and path are not enough to identify a file. For each file unique identifiers are stored to detect if a file the same file that was harvested previously.
+The input keeps a state for each file it finds. As files can be renamed or moved, the filename and path are not enough to identify a file. For each file unique identifiers are stored to detect if a file the same file that was harvested previously.
```

Comment:
This is an example of where talking about "input" as a component is a bit awkward. Wouldn't it be OK here just to say "Filebeat keeps the state for each file that it finds"?